pipelines/build/regeneration/build_job_generator.groovy

```diff
@@ -27,6 +27,9 @@ limitations under the License.
 
 String javaVersion = params.JAVA_VERSION
 String ADOPT_DEFAULTS_FILE_URL = 'https://raw.githubusercontent.com/adoptium/ci-jenkins-pipelines/master/pipelines/defaults.json'
+if (params.REPOSITORY_BRANCH != "") { // mainly for release jobs to use tag
+    ADOPT_DEFAULTS_FILE_URL = "https://raw.githubusercontent.com/adoptium/ci-jenkins-pipelines/${params.REPOSITORY_BRANCH}/pipelines/defaults.json"
```

Comment:
Do we need this if this code is on a branch? ie. line 29 can just be set to /v2023.01/....

Reply:
yes, otherwise it will still use "master" line 29's content.
and it would be good, this if--block can be eventually apply to "master" not only live on v2023.01